src/core/gui/sidebar/Sidebar.cpp

```diff
@@ -29,6 +29,8 @@ Sidebar::Sidebar(GladeGui* gui, Control* control): control(control), gui(gui), t
 
     this->sidebarContents = gui->get("sidebarContents");
 
+    // currentPage needs to be in a valid state as other functions like addPage rely on it
+    this->currentPage = this->pages.begin();
```

Comment:
Move field initialization into the declaration in the header.
Use `size_t` instead.
```suggestion
```